modules/data.land/R/extract_soil_nc.R

```diff
@@ -24,27 +24,26 @@
 #' @author Hamze Dokoohaki, Akash
 #' @export
 #'  
-extract_soil_gssurgo <- function(outdir, lat, lon, size=1, grid_size=3, grid_spacing=100, depths=c(0.15,0.30,0.60)){
+extract_soil_gssurgo <- function(outdir, lat = NULL, lon = NULL, aoi = NULL, size = 1, radius = 500, depths = c(0.15, 0.30, 0.60)){
   all.soil.ens <- list()
   
-  # create spatial bounding box
-  half_extent_m <- (grid_size - 1) / 2 * grid_spacing
-  lat_offset <- half_extent_m / 111000
-  lon_offset <- half_extent_m / (111000 * cos(lat * pi / 180))
-  
-  bbox <- sf::st_bbox(
-    c(xmin = lon - lon_offset, 
-      xmax = lon + lon_offset,
-      ymin = lat - lat_offset, 
-      ymax = lat + lat_offset),
-    crs = sf::st_crs(4326)
-  )
-  aoi <- sf::st_as_sfc(bbox)
+  # Validate inputs
+  if (is.null(aoi) && (is.null(lat) || is.null(lon))) {
+    PEcAn.logger::logger.severe("Must provide either 'aoi' OR both 'lat' and 'lon'")
+  }
   
+  # Create AOI from point + radius if needed
+  if (is.null(aoi)) {
+    aoi <- data.frame(lon = lon, lat = lat) %>%
+      terra::vect(crs = "epsg:4326") %>%
+      terra::project("epsg:5070") %>%
+      terra::buffer(width = radius) %>%
+      terra::project("epsg:4326")
```

Comment:
Reprojecting introduces distortion that can be severe, and isn't needed because terra knows how to accurately buffer in meters from a geographic CRS. Better to just pass the latlongs and let it buffer from those.
```suggestion
      terra::buffer(width = radius)
```

(for a visual demo of the distortion issue, compare the circular result of `data.frame(lat=70, lon = 0) |> vect(crs = "epsg:4326") |> buffer(1000) |> plot()` to the very elongated result of `data.frame(lat=70, lon = 0) |> vect(crs = "epsg:4326") |> project("epsg:5070") |> buffer(1000) |> project("epsg:4326") |> plot()`)

Reply:
that makes sense, removed EPSG:5070 reprojection and now apply terra::buffer() directly on the EPSG:4326 geometry